README.md

```diff
@@ -1,2 +1,35 @@
-# atomicow
-A `Cow`-like data structure where owned data is stored inside an `Arc`.
+# `atomicow`
+
+A [`Cow`](https://doc.rust-lang.org/std/borrow/enum.Cow.html)-like data structure where owned data is stored inside an [`Arc`](https://doc.rust-lang.org/std/sync/struct.Arc.html).
+Here's what it looks like:
+
+```rust, ignore
+pub enum CowArc<'a, T: ?Sized + 'static> {
+    Borrowed(&'a T),
+    Static(&'static T),
+    Owned(Arc<T>),
+}
+```
+
+As implied by the `Cow` name, this struct allows for cheap immutable reference, but can be converted into an owned form via cloning when mutation or ownership is required.
+
+This data structure is particularly useful for `str` or other values with a static lifetime,
+as might be used in structures such as asset paths.
```

Comment:
```suggestion
as it might be used in structures such as asset paths.
```
Unsure if this suggestion is correct but it feels like it is. 😅

Reply:
Nope; the "as might" is just a mildly obscure turn of phrase in English :)